### PR TITLE
Docs-978 fully branched PGD 5.6

### PR DESCRIPTION
**Pull request by djw-m**
This is a future release branch - specific instructions must be followed when merging, see "Merging this branch" section, below.
**This branch shall not be merged until PGD 5.6 is released!**

Please use this branch _only_ for changes specific to version 5.6 -
changes and corrections that apply to 5 should be made in short-lived
branches that can be merged into develop and released promptly.

## Working on this branch

Other people may be editing and altering this branch. Be sure to pull latest before
beginning work - and be aware that this branch may be rebased on a regular basis to
incorporate changes from the previous version. We recommend using `git pull --rebase` or
setting `pull.rebase` to `true` in git config to avoid mistakes,
e.g. run `git config pull.rebase true` on your local clone.

This branch will live for months, but try to merge and delete your branches within hours or days.

For changes that need review or feedback, create a branch based on this branch and
request feedback on a PR before merging your changes into this branch. 

Please use descriptive messages and comments for your commits and PRs - this allows
other people working on this PR to see at a glance what is changing.

**Good:** "updated version to 5.6 in index.mdx" / "correct misspelled option in CLI reference"

**Bad:** "updated index.mdx" / "fixed typos"

## Incorporating 5 updates

To bring in applicable changes that have been made to PGD 5, rebase
this branch against develop and force-push. Be sure to review changes for areas
that should be updated for 5.6. When resolving conflicts, defer to 5
except when changes in this branch are necessary to support 5.6.

## Merging this branch

When PGD 5.6 is released, follow this procedure to complete the separation of
documenation for 5 and 5.6:

1. Rebase one last time (see above)
2. Bring the 5 subdirectory back into the source tree:
   ```
   git checkout develop product_docs/docs/pgd/5/
   git commit -m "restore PGD 5 docs"
   git push
   ```
3. Immediately merge this branch into develop.
4. On a new branch off of develop:
   - Rewrite all `/pgd/latest/...` links in `product_docs/docs/pgd/5/**/*.mdx` to `/pgd/5/...`
   - Fix any redirect collisions that show up during a build (these will be redirects of the form `/bdr/...` that appear in both 5 and 5.6 trees)

### CODE REVIEW

**Comment by josh-heyer** [comment]
Heads up: I've slightly altered how reference index generation is invoked, to allow for multiple versions of PGD (or, future, anything else) to make use of this.

- behavior is triggered by presence of a file named `index.mdx.src`. Any directory containing such a file will (eventually) get passed to refbuilder.js
- refbuilder dependencies are initialized on process startup (right after printing the docs banner). This remains a bit ugly (should happen outside of build entirely), but at least not being invoked every createPages pass (this was the slowest part of running refbuilder previously)
- any changes to input files will still potentially trigger multiple rebuilds in develop; they should be somewhat faster though.
- refbuilder will not be invoked in CI (github actions) builds. This was only necessary in situations where files were missed anyway, and only led to issues.

**Comment by josh-heyer** [comment]
⚠Heads-up: to fix some outstanding issues caused by a merge followed by a rebase, I've re-written this branch as follows:

- PGD 5.6 is now branched from PGD 5 as of Sept 11, 2024, and incorporates all changes to PGD 5 as of that date
- Split the first commit into two: the first creates the new PGD 5.6 directory, the second rewrites links in PGD 5

This'll make it slightly easier to incorporate future PGD 5 changes if this branch persists for a while. However, if that is likely to be the case, I *strongly* recommend switching to this approach for long-lived "next version" branches: [Docs repository product version forking strategy](https://github.com/EnterpriseDB/docs/blob/develop/docs-for-docs/agreements/product-version-fork.md)

**Comment by josh-heyer** [comment]
⚠Rebased! Pull latest before continuing work. See updated description (above) for details on how this branch should be handled before and during the PGD 5.6 release. Refer to https://github.com/EnterpriseDB/docs/blob/develop/docs-for-docs/agreements/product-version-fork.md for details.

**Review by jpe442 — APPROVED** [review]
LGTM, just needs release notes looks like

File: product_docs/docs/pgd/5.6/nodes/subscriber_only/overview.mdx [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+While many use cases rely on accessing a database node which can handle queries and updates, there are also use cases which only require access to a node that can handle read-only database queries.
+Read scaling like this, by moving the read-only traffic away from active database nodes in the cluster, can improve the performance of the core cluster, whilst making database access more widely available.
+
+The basic idea of subscriber-only nodes is to provide a read-only node that can be used to offload read-only queries from the main cluster. 
+How it connects to the other nodes in the cluster can be different from how other nodes connect to each other.
+The default topology of a PGD cluster is what we call a full mesh topology, where every node connects to every other node.
+This is the most robust and fault-tolerant way to connect nodes, but it can be inefficient for some use cases.
+
+Each node in a subscriber-only group will, by default, receive replication updates from all other nodes in the cluster.
+
```
Comment (sumarjee) [inline product_docs/docs/pgd/5.6/nodes/subscriber_only/overview.mdx:17]: What is "this" on line 18th referring to?

Are these 18,19,20 lines explaining the leader based replication for SO group?
Then it again mentions similar things on line 22 but with group leader and efficient model, so a bit confusing for me.

File: product_docs/docs/pgd/5.6/nodes/subscriber_only/optimizing-so.mdx [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+With PGD 5.6 and later, each subscriber-only group gets assigned a group leader of its own. 
+This is because subscriber-only groups don't have a group Raft consensus mechanism of their own. 
+Instead, the cluster’s top level group uses its Raft consensus mechanism to handle selection of each subscriber-only group’s group leader.
+This group leader selection is on by default on PGD 5.6, regardless of the topology optimization settings.
+
+Group leaders in subscriber-only groups are regularly tested for connectivity and if unavailable, the voting nodes of top-level group select a new subscriber-only node from the subscriber-only group to become group leader. 
+The new group leader is then selected. 
+
+With optimized technology turned off, this election has no effect on the replication topology. Without the optimized topology, all nodes replicates changes to all other nodes in the  cluster.
```
Comment (sumarjee) [inline product_docs/docs/pgd/5.6/nodes/subscriber_only/optimizing-so.mdx:41]: Change: 
Without the optimized topology, all nodes replicates changes to all other nodes in the cluster
To:
Without the optimized topology, all data nodes replicates changes to all other nodes in the cluster

File: product_docs/docs/pgd/5.6/nodes/subscriber_only/overview.mdx (hunk 1 above)
Comment (djw-m) [reply product_docs/docs/pgd/5.6/nodes/subscriber_only/overview.mdx:24]: Fixed

File: product_docs/docs/pgd/5.6/nodes/subscriber_only/optimizing-so.mdx (hunk 2 above)
Comment (djw-m) [reply product_docs/docs/pgd/5.6/nodes/subscriber_only/optimizing-so.mdx:41]:
```suggestion
With optimized technology turned off, this election has no effect on the replication topology. Without the optimized topology, all data nodes replicate changes to all other nodes in the cluster.
```

File: product_docs/docs/pgd/5.6/nodes/subscriber_only/optimizing-so.mdx [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+## Group leaders in the optimized topology
+
+ With the optimized topology enabled, only the group leader receives changes from other nodes in the cluster and it takes on the responsibility of replicating those changes to the other nodes in the subscriber-only group.
```
Comment (sumarjee) [inline product_docs/docs/pgd/5.6/nodes/subscriber_only/optimizing-so.mdx:45]: It is not clear which group leader is referred here.

File: product_docs/docs/pgd/5.6/nodes/subscriber_only/optimizing-so.mdx [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+## Group leaders in the optimized topology
+
+ With the optimized topology enabled, only the group leader receives changes from other nodes in the cluster and it takes on the responsibility of replicating those changes to the other nodes in the subscriber-only group. 
+
+The other voting nodes choose the group leader from a subscriber-only group's nodes.
+Once selected, the whole cluster becomes aware of the change and any data group's write leaders then replicate data only to this newly selected group leader node. 
+Other data nodes in the data groups don't replicate data to the subscriber-only group's nodes.
+
+This approach avoids the explosion of connections that can happen when there are large numbers of SO nodes and reduces the amount of replication traffic.
```
Comment (sumarjee) [inline product_docs/docs/pgd/5.6/nodes/subscriber_only/optimizing-so.mdx:51]: In general connectivity between all nodes is required as any node can become a leader. 
Hence should we use wording as "explosion of active connections" or "connections in use". Checking with Vaijayanti too on this.

File: product_docs/docs/pgd/5.6/nodes/subscriber_only/optimizing-so.mdx (hunk 3 above)
Comment (djw-m) [reply product_docs/docs/pgd/5.6/nodes/subscriber_only/optimizing-so.mdx:45]:
```suggestion
 With the optimized topology enabled, only the subscriber-only group's group leader receives changes from other nodes in the cluster and it takes on the responsibility of replicating those changes to the other nodes in the subscriber-only group. 
```

**Comment by sumarjee** [comment]
Completed review. Shall check on updates if any.

File: product_docs/docs/pgd/5.6/nodes/subscriber_only/optimizing-so.mdx (hunk 4 above)
Comment (djw-m) [reply product_docs/docs/pgd/5.6/nodes/subscriber_only/optimizing-so.mdx:51]:
```suggestion
This approach avoids the explosion of active connections that can happen when there are large numbers of SO nodes and reduces the amount of replication traffic.
```

File: product_docs/docs/pgd/5.6/nodes/subscriber_only/optimizing-so.mdx [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+This group leader selection is on by default on PGD 5.6, regardless of the topology optimization settings.
+
+Group leaders in subscriber-only groups are regularly tested for connectivity and if unavailable, the voting nodes of top-level group select a new subscriber-only node from the subscriber-only group to become group leader. 
+The new group leader is then selected. 
+
+With optimized technology turned off, this election has no effect on the replication topology. Without the optimized topology, all data nodes replicate changes to all other nodes in the cluster.
+
+## Group leaders in the optimized topology
+
+ With the optimized topology enabled, only the subscriber-only group's group leader receives changes from other nodes in the cluster and it takes on the responsibility of replicating those changes to the other nodes in the subscriber-only group.
```
Comment (sumarjee) [inline product_docs/docs/pgd/5.6/nodes/subscriber_only/optimizing-so.mdx:45]:
```suggestion
 With the optimized topology enabled, only the subscriber-only group's group leader receives changes from other data group's write leaders in the cluster and it takes on the responsibility of replicating those changes to the other nodes in the subscriber-only group. 
```

**Pull request merged**